pyk/src/pyk/konvert/_module_to_kore.py

```diff
@@ -940,13 +985,80 @@ def _transform_rule(self, rule: KRule) -> KRule:
 
 
 @dataclass
-class AddAnywhereAtts(KompilerPass):
-    """Add the anywhere attribute to all symbol productions that are overloads or have a corresponding anywhere rule."""
+class AddImpureAtts(SingleModulePass):
+    """Add the `impure` attribute to all function symbol productions whose definition transitively contains `impure`."""
```

Comment:
While reading through this the question arose in my head whether this is really a "single-module" property.
Maybe it is only single-module because the compiler is smashing all user code into a single module? Because if imports are admitted one could easily import an `impure` "function" from a different module and call it on the RHS of a rule, or not?

Reply:
Yes, this is not single module in that sense. The name is probably misleading, but `SingleModulePass` means that it transforms a definition with a single module into another definition with a single module.